implementations/rust/ockam/ockam_entity/src/credential/error.rs

```diff
@@ -5,7 +5,7 @@ use ockam_core::Error;
 #[derive(Clone, Copy, Debug)]
 pub enum CredentialError {
     /// Mismatched number of attributes in schema and provided claims to be signed
-    MismatchedAttributesAndClaims = 1,
+    MismatchedAttributesAndClaims,
```

Comment:
@wkk could you please elaborate why do we need to change this here?

Reply:
To use custom discriminant values with an enumeration, it's required that none of the variants of the enum has associated data/fields. `CredentialError::InvalidCredentialPresentation(u32)` has a `u32` field so if we try to compile with the custom discriminant value there will be the following error:

```
error[E0658]: custom discriminant values are not allowed in enums with tuple or struct variants
  --> implementations/rust/ockam/ockam_entity/src/credential/error.rs:8:37
   |
8  |     MismatchedAttributesAndClaims = 1,
   |                                     ^ disallowed custom discriminant
...
22 |     InvalidCredentialPresentation(u32),
   |     ---------------------------------- tuple variant defined here
   |
   = note: see issue #60553 <https://github.com/rust-lang/rust/issues/60553> for more information
```

Reference: https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-field-less-enumerations

Reply:
Also I don't think the custom discriminant value is used anywhere. There is already an `as_u32` function defined for the `CredentialError` struct. So I think it should be safe to remove this.

Reply:
@wkk sure, you can delete it